File: 3498-reverse-degree-of-a-string/3498-reverse-degree-of-a-string.java
class Solution {
    public int reverseDegree(String s) {
        int n = s.length();

        int ans = 0;

        for (int i = 0; i < n; i++) {
            int idx = 'z' - s.charAt(i);

            ans = ans + (i + 1) * (idx + 1);
        }

        return ans;
    }
}

File: 3498-reverse-degree-of-a-string/3498-reverse-degree-of-a-string.cpp
class Solution {
public:
    int reverseDegree(string s) {
        // vector<int> freq(26, 0);
        int size = s.size();

        int ans=0;

        for(int i=0;i<size;i++){
            int idx='z'-s[i];

            ans+=(i+1)*(idx+1);
        }

        return ans;
    }
};